lib/pure/os.nim

```diff
@@ -1648,6 +1648,31 @@ proc setFilePermissions*(filename: string, permissions: set[FilePermission],
       var res2 = setFileAttributesA(filename, res)
     if res2 == - 1'i32: raiseOSError(osLastError(), $(filename, permissions))
 
+proc isAdmin*: bool {.noWeirdTarget.} =
+  ## Returns whether the caller's process is a member of the Administrators local
+  ## group (on Windows) or a root (on POSIX), via `geteuid() == 0`.
+  when defined(windows):
+    # Rewrite of the example from Microsoft Docs:
+    # https://docs.microsoft.com/en-us/windows/win32/api/securitybaseapi/nf-securitybaseapi-checktokenmembership#examples
+    var b: WINBOOL
+    var ntAuthority = SID_IDENTIFIER_AUTHORITY(value: [
```

Comment:
add a doc comment specifying where you got this definition; 
the above docs only shows:
`SID_IDENTIFIER_AUTHORITY NtAuthority = SECURITY_NT_AUTHORITY;`

Reply:
I copied the definition from MigGW sources, now everything should be clear.

Reply:
You can also check: https://doxygen.postgresql.org/win32security_8c.html#ae6b61e106fa5d6c5d077a9d14ee80569